File: eid-trust-service-sdk-dotnet/src/main/eid-trust-service-sdk-dotnet/eid-trust-service-sdk-dotnet/src/main/cs/ValidationFailedException.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace eid_trust_service_sdk_dotnet
{
    public class ValidationFailedException : Exception
    {
        private LinkedList<String> reasonURIs;

        public ValidationFailedException(LinkedList<String> reasonURIs)
        {
            this.reasonURIs = reasonURIs;
        }

        /// <summary>
        /// Returns the XKMS v2.0 reason URIs for the failed validation
        /// <see cref="http://www.w3.org/TR/xkms2/#XKMS_2_0_Section_5_1"/>
        /// </summary>
        /// <returns></returns>
        public LinkedList<String> getReasonURIs()
        {
            return this.reasonURIs;
        }
    }
}

File: eid-trust-service-client/src/main/java/be/fedict/trust/client/exception/ValidationFailedException.java
/*
 * eID Trust Service Project.
 * Copyright (C) 2009-2010 FedICT.
 *
 * This is free software; you can redistribute it and/or modify it
 * under the terms of the GNU Lesser General Public License version
 * 3.0 as published by the Free Software Foundation.
 *
 * This software is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public
 * License along with this software; if not, see 
 * http://www.gnu.org/licenses/.
 */

package be.fedict.trust.client.exception;

import java.util.List;

public class ValidationFailedException extends Exception {

	private static final long serialVersionUID = 1L;

	private final List<String> reasonURIs;

	public ValidationFailedException(List<String> reasonURIs) {
		this.reasonURIs = reasonURIs;
	}

	/**
	 * Returns the XKMS v2.0 reason URIs for the failed validation.
	 * 
	 * @see <a href="http://www.w3.org/TR/xkms2/#XKMS_2_0_Section_5_1">XKMS
	 *      2.0</a>
	 */
	public List<String> getReasons() {
		return this.reasonURIs;
	}
}
